design/csi-snapshots.md

```diff
@@ -0,0 +1,193 @@
+# CSI Snapshot Support
+
+Status: Draft
+
+One to two sentences that describes the goal of this proposal.
+The reader should be able to tell by the title, and the opening paragraph, if this document is relevant to them.
+
+The Container Storage Interface (CSI) has [introduced an alpha snapshot API in Kubernetes v1.12][1].
+Current plans indicate it will reach beta support in Kubernetes v1.16.
+This document suggests an approach for integrating support for this snapshot API within Velero, augmenting its existing capabilities.
+
+## Goals
+
+- Enable Velero to create CSI snapshot objects
```

Comment:
I'd reword to something like "Enable Velero to back up and restore CSI persistent volumes using the CSI snapshot API" for clarity